### PR TITLE
Added cross section tool

### PR DESCRIPTION
**Pull request by OsiJr**
udSDK update fixes [AB#2122](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/2122)

### CODE REVIEW

File: src/scene/tools/vcAddSimpleCrossSection.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    pSceneItem->HandleSceneEmbeddedUI(pProgramState);
+  }
+}
+
+void vcAddSimpleCrossSection::HandlePicking(vcState *pProgramState, vcRenderData & /*renderData*/, const vcRenderPickResult & /*pickResult*/)
+{
+  if (!pProgramState->pActiveViewport->isMouseOver)
+    return;
+
+  if (pProgramState->sceneExplorer.clickedItem.pItem != nullptr && udStrEqual(pProgramState->sceneExplorer.clickedItem.pItem->itemtypeStr, "XSlice"))
```
Comment (samsinsane) [inline src/scene/tools/vcAddSimpleCrossSection.cpp:27]: `XSlice`? Shouldn't it be called `XSection`?

File: src/scene/tools/vcAddSimpleCrossSection.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#ifndef vcAddCylinderFilter_h__
+#define vcAddCylinderFilter_h__
```
Comment (samsinsane) [inline src/scene/tools/vcAddSimpleCrossSection.h:2]: This wasn't updated.

File: src/scene/tools/vcAddSimpleCrossSection.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  void SceneUI(vcState *pProgramState) override;
+
+  void HandlePicking(vcState *pProgramState, vcRenderData &renderData, const vcRenderPickResult &pickResult) override;
+  void PreviewPicking(vcState *pProgramState, vcRenderData &renderData, const vcRenderPickResult &pickResult) override;
+
+private:
+  static vcAddSimpleCrossSection m_instance;
+};
+
+#endif //vcAddCylinderFilter_h__
```
Comment (samsinsane) [inline src/scene/tools/vcAddSimpleCrossSection.h:21]: Nor was this.

File: src/vcSettings.cpp [hunk 4]
```diff
@@ -518,6 +518,7 @@ bool vcSettings_Load(vcSettings *pSettings, bool forceReset /*= false*/, vcSetti
     vcHotkey::Set(vcB_AddBoxFilter, data.Get("keys.%s", vcHotkey::GetBindName(vcB_AddBoxFilter)).AsInt(SDL_SCANCODE_I));
     vcHotkey::Set(vcB_AddSphereFilter, data.Get("keys.%s", vcHotkey::GetBindName(vcB_AddSphereFilter)).AsInt(SDL_SCANCODE_J));
     vcHotkey::Set(vcB_AddCylinderFilter, data.Get("keys.%s", vcHotkey::GetBindName(vcB_AddCylinderFilter)).AsInt(SDL_SCANCODE_K));
+    vcHotkey::Set(vcB_AddSimpleCrossSection, data.Get("keys.%s", vcHotkey::GetBindName(vcB_AddSimpleCrossSection)).AsInt(SDL_SCANCODE_L | vcMOD_Ctrl));
```
Comment (samsinsane) [inline src/vcSettings.cpp:521]: The modifier is specified first in the others, should probably do that for consistency.

File: src/vcState.h [hunk 5]
```diff
@@ -90,6 +90,7 @@ enum vcActiveTool
   vcActiveTool_AddBoxFilter, // Add box filter in the current mouse position
   vcActiveTool_AddSphereFilter,// Add sphere filter in the current mouse position
   vcActiveTool_AddCylinderFilter,// Add cylinder filter in the current mouse position
+  vcActiveTool_AddSimpleCrossSection,// Add simple cross simple filter in the current mouse position
```
Comment (samsinsane) [inline src/vcState.h:93]: Should be `simple cross section filter` not `simple cross simple filter`.

**Review by samsinsane — CHANGES_REQUESTED** [review]

File: src/scene/tools/vcAddSimpleCrossSection.cpp (hunk 1 above)
Comment (OsiJr) [reply src/scene/tools/vcAddSimpleCrossSection.cpp:27]: r/7characterlimit

Comment (samsinsane) [reply src/scene/tools/vcAddSimpleCrossSection.cpp:27]: Of course, I forgot the limit is really low.

**Review by samsinsane — APPROVED** [review]

**Pull request merged**